core-java-modules/core-java-streams-maps/src/test/java/com/baeldung/streams/tomapvsgroupingby/ToMapVsGroupingByUnitTest.java

```diff
@@ -0,0 +1,203 @@
+package com.baeldung.streams.tomapvsgroupingby;
+
+import static com.google.common.collect.Lists.newArrayList;
+import static com.google.common.collect.Sets.newHashSet;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+
+class City {
+
+    private String name;
+    private String country;
+
+    public City(String name, String country) {
+        this.name = name;
+        this.country = country;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    @Override
+    public final boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof City city)) {
+            return false;
+        }
+
+        return Objects.equals(name, city.name) && Objects.equals(country, city.country);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = Objects.hashCode(name);
+        result = 31 * result + Objects.hashCode(country);
+        return result;
+    }
+}
+
+public class ToMapVsGroupingByUnitTest {
+
+    private City paris = new City("Paris", "France");
+    private City berlin = new City("Berlin", "Germany");
+    private City london = new City("London", "UK");
+    private City newYork = new City("New York", "USA");
+    private City tokyo = new City("Tokyo", "Japan");
+
+    private City nice = new City("Nice", "France");
+    private City hamburg = new City("Hamburg", "Germany");
+    private City aachen = new City("Aachen", "Germany");
+
+    private City franceNull = new City(null, "France");
+    private City germanyNull = new City(null, "Germany");
```

Comment:
Can we make these private static final and uppercase if we are using them as constants? Or at least private final

Reply:
OK, I will make them final. They can be static for sure, but IMO, lowercase words are easier to read than UPPERCASE words. 
I will make the change f together with your article suggestions.

Reply:
I agree, but technically we are supposed to be following https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names for constant naming, and the City type looks like an immutable class

I'm okay with lowercase final personally, not sure if the senior review will be picky about this or not

Reply:
Yes, technically they should be static variables. Ok, I will change them into "final static". 👍